src/utils/Disposables.ts

```diff
@@ -14,24 +14,25 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-function disposeValue(value) {
+type Func = () => void;
+type Disposable = { dispose: Func; [key: string]: any } | Func;
```

Comment:
why do you need the `[key: string]: any`? We might want to use an IDisposable interface in the future, so could this:
```suggestion
export interface IDisposable {
    dispose(): void;
}

type Disposable = IDisposable | (() => void);
```